lib/parsing.js

```diff
@@ -0,0 +1,235 @@
+"use strict";
+
+const { RequestError, ConfigurationError } = require("./errors");
+
+/**
+ * A parser for the Accept header
+ * @param {Object.<string, string|undefined>} requestHeaders
+ * @param {string[]} preferences
+ */
+exports.parseAccept = (requestHeaders, preferences) => {
```

Comment:
is there an existing trusted source we can use instead of implementing this ourselves? maybe https://www.npmjs.com/package/negotiator

Reply:
giving this section a lighter review as a result of this question